lib/sse_consumer.ex

```diff
@@ -0,0 +1,150 @@
+defmodule SSEConsumer do
+  require Logger
+  use GenServer
+
+  defmodule Request do
+    @enforce_keys ~w(method url body headers)a
```

Comment:
I wonder if this is almost to generic. My suspicion is that SSE's rarely make sense unless a GET request and a GET request would never have a body.

p.s. musing not instruction

Reply:
I don't want to make that assumption right now, plus the Request would look naked without the method ;)

Reply:
fair enough